aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -134,25 +123,20 @@ synchronized void startSubscriptions() throws DataStoreException {
             .subscribe(buffer::onNext, buffer::onError, buffer::onComplete)
         );
 
-        boolean subscriptionsStarted;
         try {
             LOG.debug("Waiting for subscriptions to start.");
-            subscriptionsStarted = latch.abortableAwait(adjustedTimeoutSeconds, TimeUnit.SECONDS);
+            latch.abortableAwait();
```

Comment:
Same concern here, as stated above. This has the potential to hang the sync engine, and without any error rendered.